lib/oauth2.js

```diff
@@ -61,9 +71,14 @@ async function _getNewAccessToken({
     body.append('scopes', scopes.join(' '));
   }
   for(; maxRetries >= 0; --maxRetries) {
-    const access_token = await _requestAccessToken(
-      {url: token_endpoint, body});
+    const {
+      access_token,
+      expires_in = ttl
+    } = await _requestAccessToken({url: token_endpoint, body});
     if(access_token) {
+      // convert seconds expires_in to milliseconds ttl for cache
+      const options = {ttl: expires_in * 1000};
```

Comment:
```suggestion
      const options = {ttl: expires_in};
```

Don't do millisecond calculations inline, do it when you define/use the variable unless absolutely necessary.

Reply:
`ttl` needs to be in seconds because it has to match `expires_in` from the oauth access token request. I went ahead and just removed the global ttl, so we set ttl individually for each token based off of the `expires_in` value and if `expires_in` is not returned in the accessToken request it is set to one hour.

Reply:
Ok, I think the thing that was throwing me was that there are two meanings of "ttl" in this code... "ttl in seconds" and "ttl in milliseconds" and those two usages are within 4 lines of each other. Code smell isn't great here, but fine for now given that this isn't critical path code.